frontend/app/components/common/login/index.js

```diff
@@ -0,0 +1,99 @@
+import Component from '@glimmer/component';
+import EmberObject, { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import config from 'ui/config/environment';
+import { inject as service } from '@ember/service';
+
+export default class CommonLoginComponent extends Component {
+  @tracked loginData;
+  @tracked errorMessage;
+  @tracked formInvalid = true;
+
+  @service('auth') auth;
+
+  loginURL = config.APP.BASE_API_URL + config.APP.LOGIN_URL;
+
+  get initialValues() {
+    return {
+      email: { value: '', error: '' },
+      password: { value: '', error: '' },
+    };
+  }
+
+  constructor() {
+    super(...arguments);
+    this.loginData = EmberObject.create(this.initialValues);
+  }
+
+  @action
+  save(event) {
+    this.errorMessage = null;
+    event.preventDefault();
+    this.#login();
+  }
+
+  @action
+  reset() {
+    this.errorMessage = null;
+    this.loginData.setProperties(this.initialValues);
+  }
+
+  @action
+  validateRequiredFields(id) {
+    const setValue = (id) => {
+      const value = this.loginData.get(id).value;
+
+      let errorMsg = '';
+      if (!value) {
+        errorMsg = 'Required Input';
+      }
+      if (id === 'email' && !this.#validateEmail(value)) {
+        errorMsg = 'Invalid Email';
+      }
+
+      this.loginData.set(id, {
+        value: value,
+        error: errorMsg,
+      });
+    };
+    if (id) {
+      setValue(id);
+    } else {
+      const fieldNames = Object.keys(this.loginData);
+      fieldNames.forEach((field) => {
+        setValue(field);
+      });
+    }
+    this.setFormValid();
+  }
+
+  onLoginSuccess(user) {
+    this.args.onLoginSuccess(user);
+  }
+
+  setFormValid() {
+    const isValid = Object.keys(this.loginData).every((field) => {
+      return this.loginData[field].value && !this.loginData[field].error;
+    });
+    this.formInvalid = isValid ? null : true;
```

Comment:
why null?. why not !isValid

Reply:
this flag is used to remove or set the 'disabled' attribute in button. in case to remove disabled attr null has to be passed in template. !valid will still set the attribute with false value